### DISCUSSION TITLE
Image orientation not correctly detected

### DISCUSSION THREAD

**Discussion by ep0p** [mindee/doctr#1881] [Q&A]
Hi,

I'm working on an OCR pipeline where I need to detect image orientation before applying layout tasks. I'm using Doctr for this, but I'm getting inconsistent results with some vertical images. For example, the attached image is returning an orientation angle of 180°.

Any insights on what might be going wrong?

![corrected_page_1_v1](https://github.com/user-attachments/assets/426c9b94-0914-414c-9e97-147dfe1b3220)

**Comment by felixdittrich92** [comment]
Hi @ep0p :wave:,

We also saw such cases, the only and easiest way here is to fine tune the model .. luckily everything you need are plain images (without rotations) because the labels are augmented so that's really straight forward :)

**Reply by ep0p** [reply]
Hi @felixdittrich92,

Thanks a lot for you answer, i'll do the fine tuning then.

**Reply by ep0p** [reply]
@felixdittrich92 do you have any recommendation for a high-performing classification model that I should tune?

**Reply by ep0p** [reply]
scrap that, I think found it: page_orientation_predictor() :D

**Reply by felixdittrich92** [reply]
:D

Yeah currently we support only:
`mobilenet_v3_small_page_orientation`
and
`mobilenet_v3_small_crop_orientation`
as architecture for this task

Additional you can modify the auto augmentation part to your needs, currently we additional augment in the range (-180, 180) so full rotation

https://github.com/mindee/doctr/blob/225210c2536662fee7e3989932833833a7eb0f09/references/classification/train_pytorch_orientation.py#L44

if you only have to deal with exact [-90, 0, 90, 180] cases you could comment the following part:
```python
# augment the angle randomly with a probability of 0.5
    if np.random.rand() < 0.5:
        angle += float(np.random.choice(np.arange(-25, 25, 5)))
```

**Reply by ep0p** [reply]
hmmm i just finished 3 fine-tuning attempts, with a ds of 300images/train and 100images/val but the results are not that good; i must be doing something wrong 

```
python references/classification/train_pytorch_orientation.py mobilenet_v3_small_page_orientation --type page --train_path /home/epop/DATASET/GOT/output_dataset_20k/train --val_path /home/epop/DATASET/GOT/output_dataset_20k/val --name mobilenet_v3_small_page_orientation_custom_v3 --epochs 20 --b 256 --early-stop
```
the learning rate is decreasing properly, but the accuracy is fluctuating and never depassing 35%
i have tried 20, 30 , 50 epochs

**Reply by felixdittrich92** [reply]
@ep0p `--pretrained` missing :D

**Reply by ep0p** [reply]
i have confirmed, again, the problem is in chair not in computer
thanks a lot!
